.build-tools/cmd/zz-e2e-perf.go

```diff
@@ -417,25 +423,98 @@ func (c *cmdE2EPerf) getHashDir() (string, error) {
 			return nil
 		}
 
-		// Compute the sha256 of the file
-		checksum, err := checksumFile(path)
+		// Add the hash of the file
+		checksum, err := hashEntryForFile(path, relPath)
 		if err != nil {
 			return err
 		}
 
-		// Convert all slashes to / so the hash is the same on Windows and Linux
-		relPath = filepath.ToSlash(relPath)
-
-		files = append(files, relPath+" "+checksum)
+		files = append(files, checksum)
 		return nil
 	})
+
+	if err != nil {
+		return nil, err
+	}
+	return files, nil
+}
+
+func hashEntryForFile(path string, relPath string) (string, error) {
+	// Compute the sha256 of the file
+	checksum, err := checksumFile(path)
+	if err != nil {
+		return "", err
+	}
+
+	// Convert all slashes to / so the hash is the same on Windows and Linux
+	relPath = filepath.ToSlash(relPath)
+
+	return relPath + " " + checksum, nil
+}
+
+// Returns the checksum of the files in the directory
+func (c *cmdE2EPerf) getHashDir() (string, error) {
+	basePath := filepath.Join(c.getAppDir(), c.flags.Name)
+	_, err := os.Stat(basePath)
+	if err != nil {
+		fmt.Printf("could not find app %s\n", basePath)
+		return "", err
+	}
+
+	// Load the files to exclude
+	ignores := c.getIgnores()
+
+	// Load additional paths to include
+	includes := c.getIncludes()
+
+	// Compute the hash of the app's files
+	files, err := hashFilesInDir(basePath, ignores)
 	if err != nil {
 		return "", err
 	}
 	if len(files) == 0 {
 		return "", fmt.Errorf("no file found in the folder")
 	}
 
+	// Include other files or folders as needed
+	if len(includes) > 0 {
```

Comment:
I guess you don't need to check for length as if includes are empty so the loop will not be executed, right?

Reply:
`includes` could be `nil` so this is actually a nil-check

Reply:
I was not sure about that so I tested it out https://go.dev/play/p/ar7iqfsZtpR

and you can actually iterate over nil arrays which won’t result in a nil dereference panic

Reply:
Done